### PR TITLE
Match Interface_DrawClock

### PR DESCRIPTION
**Pull request by engineer124**
Finally!

### CODE REVIEW

File: src/code/z_parameter.c [hunk 1]
```diff
@@ -4366,527 +4364,489 @@ void Interface_DrawClock(PlayState* play) {
...
-                    gDPSetCombineLERP(OVERLAY_DISP++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE,
-                                      TEXEL0, 0, PRIMITIVE, 0);
+            //! @bug A texture height of 50 is given below. The texture is only 48 units height
+            //!      resulting in this reading into the next texture. This results in a white
+            //!      dot in the bottom center of the clock. For the three-day clock, this is
+            //!      covered by the diamond. However, it can be seen by the final-hours clock.
+            OVERLAY_DISP = Gfx_DrawTexRect4b(OVERLAY_DISP, gThreeDayClockBorderTex, 4, 64, 50, 96, 168, 128, 50, 1, 6,
+                                             0, 1 << 10, 1 << 10);
+
+            if (((CURRENT_DAY >= 4) || ((CURRENT_DAY == 3) && (((void)0, gSaveContext.save.time) >= 5) &&
```
Comment (hensldm) [inline src/code/z_parameter.c:4450]:
```suggestion
            if (((CURRENT_DAY >= 4) || ((CURRENT_DAY == 3) && (((void)0, gSaveContext.save.time) >= (TIME_CLOCK(0,0) + 5) &&
```
Not 100% sure 🤷.

File: src/code/z_parameter.c [hunk 2]
```diff
@@ -4366,527 +4364,489 @@ void Interface_DrawClock(PlayState* play) {
...
-                            colorStep =
-                                ABS_ALT(sFinalHoursClockFrameEnvGreen - D_801BFC0C[sFinalHoursClockColorTargetIndex]) /
-                                sFinalHoursClockColorTimer;
-                            if (sFinalHoursClockFrameEnvGreen >= D_801BFC0C[sFinalHoursClockColorTargetIndex]) {
-                                sFinalHoursClockFrameEnvGreen -= colorStep;
-                            } else {
-                                sFinalHoursClockFrameEnvGreen += colorStep;
-                            }
+            // Final Hours
+            if ((CURRENT_DAY >= 4) || ((CURRENT_DAY == 3) && (((void)0, gSaveContext.save.time) >= 5) &&
```
Comment (hensldm) [inline src/code/z_parameter.c:4711]: Same comment as above.

File: src/code/z_parameter.c [hunk 3]
```diff
@@ -4366,527 +4364,489 @@ void Interface_DrawClock(PlayState* play) {
...
-                    gSP1Quadrangle(OVERLAY_DISP++, 4, 6, 7, 5, 0);
+            // Colours the Three-Day Clocks's Hour Digit Above the Sun
+            gDPPipeSync(OVERLAY_DISP++);
+            gDPSetPrimColor(OVERLAY_DISP++, 0, 0, 255, 255, 155, sThreeDayClockAlpha);
+            gSP1Quadrangle(OVERLAY_DISP++, 4, 6, 7, 5, 0);
+
+            /**
+             * Draws Three-Day Clock's Hour Digit Above the Moon
+             */
+            // Rotates Three-Day Clock's Hour Digit To Above the Moon
```
Comment (jpburnett) [inline src/code/z_parameter.c:4688]: Nit: Is there a specific reason to have the different style of commenting? could these be combined?

**Review by hensldm — APPROVED** [review]
1 possible suggestion

**Review by jpburnett — APPROVED** [review]
Mainly only nits on the inconsistent use of commenting style (// vs /**/) but nothing that I think would prevent an approval.

**Review by AngheloAlf — APPROVED** [review]
Looks good besides the comments from other reviews

File: src/code/z_parameter.c (hunk 1 above)
Comment (engineer124) [reply src/code/z_parameter.c]: Yeah sure, I suppose that makes sense

File: src/code/z_parameter.c (hunk 3 above)
Comment (engineer124) [reply src/code/z_parameter.c]: The idea I was trying to go for was that this function is incredibly long (500+ lines in C), so the /* comments are for describing entire blocks of code (like sections or chapter titles), while the // comments are for describing 1-2 line comments. I think this is the only place where there's both which is why it's confusing. I put an extra line gap between them, but other suggestions are welcome

Comment (hensldm) [reply src/code/z_parameter.c]: I might add a `section:`/`block:` for the /* comments to help clarify.

Comment (engineer124) [reply src/code/z_parameter.c]: I put in `Section: ` for them

**Pull request merged**